packages/foam-core/src/markdown-provider.ts

```diff
@@ -91,7 +91,7 @@ export function createMarkdownReferences(
       return {
         linkText: link.to,
         wikiLink: pathToNote,
-        pageTitle: target.title,
+        pageTitle: target.title || target.id,
```

Comment:
This is the only place where `note.title` was used directly.  `pageTitle` is initialized by `note.id` if heading doesn't exist.